File: playwright-csharp/src/Diagnostics/ApiEvidence.cs
namespace TestTracks.Playwright.CSharp.Diagnostics;

/// <summary>
/// Captures API request evidence that can be written into the test report.
/// </summary>
public sealed class ApiEvidence
{
    private readonly List<string> _entries = [];

    public IReadOnlyList<string> Entries => _entries;

    /// <summary>
    /// Records an API operation with the response status and a redacted response body.
    /// </summary>
    public void Record(string operation, int status, string? body)
    {
        _entries.Add($"{operation} -> HTTP {status}{Environment.NewLine}{SecretRedactor.Redact(body)}");
    }

    public string Text()
    {
        return _entries.Count == 0
            ? "No API evidence was captured."
            : string.Join($"{Environment.NewLine}{Environment.NewLine}", _entries);
    }
}

File: playwright-java/src/main/java/testtracks/diagnostics/ApiEvidence.java
package testtracks.diagnostics;

import java.util.ArrayList;
import java.util.List;

public final class ApiEvidence {
  private final List<String> entries = new ArrayList<>();

  public List<String> entries() {
    return List.copyOf(entries);
  }

  public void record(String operation, int status, String body) {
    entries.add(
        operation + " -> HTTP " + status + System.lineSeparator() + SecretRedactor.redact(body));
  }

  public String text() {
    return entries.isEmpty()
        ? "No API evidence was captured."
        : String.join(System.lineSeparator() + System.lineSeparator(), entries);
  }
}
